docs/docs/reference-react-component.md

```diff
@@ -230,34 +230,59 @@ componentWillUnmount()
 ### `setState()`
 
 ```javascript
-setState(nextState, callback)
+setState(updater, [callback])
 ```
 
-Performs a shallow merge of nextState into current state. This is the primary method you use to trigger UI updates from event handlers and server request callbacks.
+`setState()` is an asynchronous method which enqueues changes to be made to the state during the next update cycle. This is the primary method you use to trigger updates in response to event handlers, server responses, etc...
```

Comment:
> `setState()` is an asynchronous method

Sometimes it's sync and sometimes it's async, depending on the priority of the update.

> during the next update cycle

What does "update cycle" mean? The next time the component is rendered? That's not true because the update might not have enough priority.